cpp/src/arrow/flight/sql/driver.cc

```diff
@@ -0,0 +1,1877 @@
+                                              struct AdbcPartitions* partitions,
+                                              int64_t* rows_affected,
+                                              struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementExecutePartitions(
+      statement, schema, partitions, rows_affected, error);
+}
+
+AdbcStatusCode AdbcStatementExecuteQuery(struct AdbcStatement* statement,
+                                         struct ArrowArrayStream* out,
+                                         int64_t* rows_affected,
+                                         struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementExecuteQuery(statement, out, rows_affected,
+                                                            error);
+}
+
+AdbcStatusCode AdbcStatementGetParameterSchema(struct AdbcStatement* statement,
+                                               struct ArrowSchema* schema,
+                                               struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementGetParameterSchema(statement, schema,
+                                                                  error);
+}
+
+AdbcStatusCode AdbcStatementNew(struct AdbcConnection* connection,
+                                struct AdbcStatement* statement,
+                                struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementNew(connection, statement, error);
+}
+
+AdbcStatusCode AdbcStatementPrepare(struct AdbcStatement* statement,
+                                    struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementPrepare(statement, error);
+}
+
+AdbcStatusCode AdbcStatementRelease(struct AdbcStatement* statement,
+                                    struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementRelease(statement, error);
+}
+
+AdbcStatusCode AdbcStatementSetOption(struct AdbcStatement* statement, const char* key,
+                                      const char* value, struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementSetOption(statement, key, value, error);
+}
+
+AdbcStatusCode AdbcStatementSetSqlQuery(struct AdbcStatement* statement,
+                                        const char* query, struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementSetSqlQuery(statement, query, error);
+}
+
+AdbcStatusCode AdbcStatementSetSubstraitPlan(struct AdbcStatement* statement,
+                                             const uint8_t* plan, size_t length,
+                                             struct AdbcError* error) {
+  return arrow::flight::sql::FlightSqlStatementSetSubstraitPlan(statement, plan, length,
+                                                                error);
+}
+
+extern "C" {
+ARROW_FLIGHT_SQL_EXPORT
+AdbcStatusCode AdbcDriverInit(int version, void* raw_driver, struct AdbcError* error) {
```

Comment:
This symbol is meant to be the canonical entrypoint for the driver manager; user code isn't meant to call it. The driver manager uses dlsym, so there shouldn't be a risk of other driver libraries interfering. The export is to make sure it gets the right linkage on Windows.

Reply:
Right, but that's if you want the driver manager to `dlopen` this driver. People might instead want to link against `libflight_sql` directly (because they also want to use the C++ APIs) and then call `AdbcLoadDriverFromInitFunc`.

Reply:
And then you would pass it the C++/namespaced one, no?